staging/src/k8s.io/apimachinery/pkg/util/sets/set.go

```diff
@@ -27,6 +28,12 @@ type Set[T comparable] map[T]Empty
 // cast transforms specified set to generic Set[T].
 func cast[T comparable](s map[T]Empty) Set[T] { return s }
 
+// MarshalJSON implements the json.Marshaler interface for Set.
+func (s Set[T]) MarshalJSON() ([]byte, error) {
+	items := s.UnsortedList()
```

Comment:
@thockin Do we need to use sorted list by `List(s)`? Performance PoV, `UnsortedList` makes sense. But for consistency, it would be better to use `List(s)`.

Please let me know which aspects (performance vs consistency) we should respect here.

Reply:
JSON marshal on a map will sort, so I think this _should_ be sorted.  But `type Set` has a `comparable` constraint, and `List()` only works for `ordered` args - a subset of `comparable`.

So unless there is some trick I can't think of at the moment, you HAVE TO leave it unsorted as long as it is a method.  You could change this to a free function, like `List()` but then it doesn't work for logging.

Reply:
That sounds reasonable. Thank you for clarifying the context!

Reply:
Would be good to add a comment to that effect.